src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -384,15 +414,18 @@ const UCollator* GetCollatorFromSortHandle(SortHandle* pSortHandle, int32_t opti
             assert(false && "Unexpected pthread_mutex_lock return value.");
         }
 
-        TCollatorMap::iterator entry = pSortHandle->collatorsPerOption.find(options);
-        if (entry == pSortHandle->collatorsPerOption.end())
+        TCollatorMap* map = (TCollatorMap*)malloc(sizeof(TCollatorMap));
```

Comment:
I cannot see the map being deleted, am I missing something?

Reply:
Map will be deleted in `CloseSortHandle` in `if` case (`tsearch()` is basically search-and-insert, and tfind is just search..  `man tsearch` ), in `else` case  we can free it.